### PR TITLE
Map k8s service ports to values

### LINKED ISSUE #22327: k8s service ports don't reflect in proxy ports for gateways
Values.yaml port settings in gateways map not just to service but also to deployment (for the gateway proxy container). The current k8s overlay maps only to the service.

### PR DESCRIPTION
**Pull request by ostromart**
Fixes https://github.com/istio/istio/issues/22327.

### CODE REVIEW

File: operator/pkg/component/component.go [hunk 1]
```diff
@@ -719,6 +719,17 @@ func isCoreComponentEnabled(c *CommonComponentFields) bool {
 	return enabled
 }
 
+func getK8sResourceSpec(c *CommonComponentFields) *v1alpha1.KubernetesResourcesSpec {
+	switch t := c.spec.(type) {
+	case *v1alpha1.GatewaySpec:
+		return t.K8S
+	case *v1alpha1.ComponentSpec:
```
Comment (richardwxn) [inline operator/pkg/component/component.go]: what is the difference between these two?

**Review by richardwxn — COMMENTED** [review]
I think with this change the ports of gateways would not be able to set via values api, right? I think that is ok because it aligns with the situation of other k8s settings but we should really come up with a plan to make user aware, either documentation or something else we discussed.

File: operator/pkg/component/component.go (hunk 1 above)
Comment (ostromart) [reply operator/pkg/component/component.go]: nothing much anymore but these are just separate types to support any specializations in the future. see istio/api

**Comment by ostromart** [comment]
yes, it overrides any settings in values, just as for all other k8s spec settings.

**Review by richardwxn — APPROVED** [review]

**Comment by sdake** [comment]
hi @ayj @ostromart.

I took a look at the cherrypick as I was too late in the 5 hours the PR was open, to review this PR. I think there are some subtle potential regressions introduced with multicluster. @ayj should take a look, and I'd suggest reverting some very targeted parts of this PR (see review in cherrypick https://github.com/istio/istio/pull/22338), or make a new PR and cherrypick that in addition.

**Pull request merged**